### ISSUE TITLE
Should widget State be exposed at the top level, for custom widgets?

### ISSUE THREAD

**Issue by viperscape** [PistonDevelopers/conrod#456]
When creating a custom widget, I found a bit of trouble when trying to use ```widget::State```. Widget trait method ```update``` requires ```State``` as an argument: https://github.com/PistonDevelopers/conrod/blob/master/src/widget/mod.rs#L82
However, it's private at the top level, it seems
```
error: struct `State` is private
use conrod::widget::State; 
```

I'm following along by copying off of [Label widget](https://github.com/PistonDevelopers/conrod/blob/master/src/widget/label.rs#L80)

Thanks

**Comment by mitchmindtree** [comment]
Ahh it should indeed! thanks @viperscape , should be fixed with #457 now.

**Comment by viperscape** [comment]
great thanks! and as WidgetState, terrific
